feat: add `stats/incr/nanmhmean`

satansin123 opened the pull request:
Resolves #5577.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the @stdlib/stats/incr/nanmhmean package, which computes an incremental mean while handling NaN values by skipping them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5577

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

tests - 
![image](https://github.com/user-attachments/assets/27ee10c3-daa5-470a-92bb-4b21ddafb57d)

benchmark - 
![image](https://github.com/user-attachments/assets/2d93a189-7c10-4fd1-b989-a7804622d782)

examples - 
![image](https://github.com/user-attachments/assets/86308c8e-02b8-4e83-9806-5863f2c59a6a)


## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

satansin123 committed:
feat: add stats/incr/nanmhmean

satansin123 committed:
refractor: correct lint error

satansin123 commented:
/stdlib help

satansin123 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmhmean/README.md:
````diff
@@ -0,0 +1,160 @@
+var incrnanmhmean = require( '@stdlib/stats/incr/nanmhmean' );
+```
+
+#### incrmhmean( window )
````
Wrong function name.

```suggestion
#### incrnanmhmean( window )
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmhmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,75 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrmhmean( 3 );
```
Wrong function name.

```suggestion
* var accumulator = incrnanmhmean( 3 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmhmean/package.json:
```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@stdlib/stats/incr/nanmhmean",
+  "version": "0.0.0",
+  "description": "Compute a moving harmonic mean incrementally, ignoring `NaN` values.",
```
Don't wrap `NaN` in backticks in `package.json` descriptions.

```suggestion
  "description": "Compute a moving harmonic mean incrementally, ignoring NaN values.",
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmhmean/lib/main.js:
```diff
@@ -0,0 +1,115 @@
+	*/
+	function accumulator( x ) {
+		var v;
+		if ( arguments.length === 0 || isnan(x) ) {
```
Missing spaces around operators and after function names.
```suggestion
		if ( arguments.length === 0 || isnan( x ) ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmhmean/examples/index.js:
```diff
@@ -0,0 +1,42 @@
+		v = randu() * 100.0;
+	}
+	m = accumulator( v );
+	console.log( '%d\t%d', v.toFixed( 4 ), ( m === null ) ? NaN : m.toFixed( 4 ) );
```
`.toFixed()` returns a string, so use `%s` instead of `%d`.

Planeshifter reviewed (changes requested):
Thanks for the PR! Some late feedback if you are still interested in moving it along.

Planeshifter committed:
fix: address review feedback

- Fix function name in README heading and types example
- Remove backticks from NaN in package.json description
- Add spaces inside isnan() call in main.js
- Fix format specifiers in examples/index.js
- Use relative require path in test.js
- Update copyright years to 2026

Planeshifter commented:
/stdlib merge

Planeshifter reviewed (approved):
Thanks, @satansin123!
